core/debug/inc/debug.h

```diff
@@ -18,8 +18,19 @@
 #define __DEBUG_H__
 
 #include <uvisor.h>
+#include "debug_exports.h"
 #include "halt.h"
 
+/* Debug box handle.
+ * This is use internally by uVisor to keep track of the registered debug box
+ * and access its driver when needed. */
+typedef struct TDebugBox {
+    const TUvisorDebugDriver *driver;
+    uint8_t box_id;
+    int initialized;
+} UVISOR_PACKED TDebugBox;
+TDebugBox g_debug_box;
```

Comment:
This is a one-per-uVisor instance. Do we want to consider this being one-per-box and place it in the box context (box-local memory)? We'd also handle registration on a per-box (instead of per-uVisor) basis.